logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/SyncManager.kt

```diff
@@ -0,0 +1,53 @@
+package com.wire.kalium.logic.sync
+
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+enum class SyncState {
+    WAITING,
+    SLOW_SYNC,
+    FETCHING_CONVERSATIONS,
+    COMPLETED,
+}
+
+class SyncManager(private val conversationRepository: ConversationRepository) {
+
+    private val internalSyncState = MutableStateFlow(SyncState.WAITING)
+    private var syncJob: Job? = null
+
+    suspend fun performSlowSync() {
+        internalSyncState.update { SyncState.FETCHING_CONVERSATIONS }
+        conversationRepository.fetchConversations()
```

Comment:
Was looking at the slow sync implementation for the android reloaded and saw that each slow sync phase was implemented as separate use case instead of calling repositories directly. I guess it makes sense to adopt the same  approach here.